src/test/java/org/folio/inventory/dataimport/consumers/MarcHoldingsHridSetKafkaHandlerTest.java

```diff
@@ -0,0 +1,183 @@
+package org.folio.inventory.dataimport.consumers;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import org.folio.DataImportEventPayload;
+import org.folio.HoldingsRecord;
+import org.folio.MappingProfile;
+import org.folio.ParsedRecord;
+import org.folio.Record;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.common.domain.Success;
+import org.folio.inventory.dataimport.HoldingWriterFactory;
+import org.folio.inventory.dataimport.handlers.actions.HoldingUpdateDelegate;
+import org.folio.inventory.domain.HoldingsRecordCollection;
+import org.folio.inventory.storage.Storage;
+import org.folio.kafka.cache.KafkaInternalCache;
+import org.folio.processing.events.utils.ZIPArchiver;
+import org.folio.processing.mapping.MappingManager;
+import org.folio.processing.mapping.mapper.reader.record.marc.MarcHoldingsReaderFactory;
+import org.folio.rest.jaxrs.model.EntityType;
+import org.folio.rest.jaxrs.model.Event;
+import org.folio.rest.jaxrs.model.MappingDetail;
+import org.folio.rest.jaxrs.model.MappingRule;
+import org.folio.rest.jaxrs.model.ProfileSnapshotWrapper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+import static org.folio.ActionProfile.FolioRecord.HOLDINGS;
+import static org.folio.ActionProfile.FolioRecord.MARC_HOLDINGS;
+import static org.folio.DataImportEventTypes.DI_SRS_MARC_HOLDINGS_HOLDING_HRID_SET;
+import static org.folio.rest.jaxrs.model.ProfileSnapshotWrapper.ContentType.MAPPING_PROFILE;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.when;
+
+@RunWith(VertxUnitRunner.class)
```

Comment:
I think there is no need for this runner because you mock all needed classes and deploying verticle is not needed.

Reply:
It is not about deploying a verticle, it about supporting asynchronous unit tests using TestContext and Async objects